src/tests/venues/[id].test.jsx

```diff
@@ -20,52 +21,47 @@ const getMocks = () => [
     },
     result: {
       data: {
-        venue: {
-          addressLocality: "Helsinki",
-          dataSource: "",
-          description: "",
-          email: "",
-          id,
-          image: "",
-          infoUrl: "https://hel.fi",
-          name: "Eiran uimaranta",
-          position: {
-            type: "Point",
-            coordinates: [1, 2],
-          },
-          postalCode: "00001",
-          streetAddress: "Eirantie 3",
-          telephone: "+35812345678",
-          openingHours: getOpeningHoursForWeek(),
-          isOpen: false,
-          ontologyTree: [],
-          ontologyWords: [],
-          accessibilitySentences: [
-            {
-              groupName: "Pääsisäänkäynti",
-              sentences: ["Erottuu selkeästi", "Ahdas tuulikaappi"],
-            },
-          ],
-        },
+        venue: getVenue({ id }),
       },
     },
   },
 ];
 
-describe("venues/[id]", () => {
-  it("renders without crashing", async () => {
-    await act(async () => {
-      render(<VenuePageContent />, getMocks(), {
-        query: {
-          id,
-        },
-      });
+test("venues/[id] renders correctly", async () => {
+  await act(async () => {
+    render(<VenuePageContent />, getMocks(), {
+      query: {
+        id,
+      },
     });
-
-    await waitFor(() =>
-      expect(
-        screen.getByText("Eiran uimaranta", { selector: "h2" })
-      ).toBeInTheDocument()
-    );
   });
+
+  await waitFor(() =>
+    expect(
+      screen.getByText("Eiran uimaranta", { selector: "h2" })
+    ).toBeInTheDocument()
+  );
+
+  // -- Check that the user can access price information correctly
+  const allPriceConnectionsContentLines = defaultConnections
+    .filter((item) => item.sectionType === "PRICE")
+    .map((item) => item.name)
+    .join("\n\n")
+    .split("\n");
+  const startOfFirstPriceConnection = allPriceConnectionsContentLines
+    .slice(0, 3)
+    // Testing library does some formatting to text content which removes line
+    // changes. Matching with them fails, matching without works.
+    .join("");
+  // User sees the beginning of the price
+  expect(screen.getByText(startOfFirstPriceConnection)).toBeInTheDocument();
```

Comment:
When using `toBeInTheDocument` assertion I think `queryBy*` selector would be better. When using `getBy` and element is not found, it already throws error and assertion is never done.

Reply:
Either drop the assertion and leave only selector

`screen.getByText(startOfFirstPriceConnection)`

or use queryByText

`expect(screen.queryByText(startOfFirstPriceConnection)).toBeInTheDocument();`

Not critical or a must change but just my opinion :)